miio/airpurifier_miot.py

```diff
@@ -0,0 +1,405 @@
+
+    @property
+    def motor_speed(self) -> int:
+        """Speed of the motor."""
+        return self.data["motor_speed"]
+
+    @property
+    def filter_rfid_product_id(self) -> Optional[str]:
+        """RFID product ID of installed filter."""
+        return self.data["filter_rfid_product_id"]
+
+    @property
+    def filter_rfid_tag(self) -> Optional[str]:
+        """RFID tag ID of installed filter."""
+        return self.data["filter_rfid_tag"]
+
+    @property
+    def filter_type(self) -> Optional[FilterType]:
+        """Type of installed filter."""
+        if self.filter_rfid_tag is None:
+            return None
+        if self.filter_rfid_tag == "0:0:0:0:0:0:0":
+            return FilterType.Unknown
+        if self.filter_rfid_product_id is None:
+            return FilterType.Regular
+        return self._get_filter_type(self.filter_rfid_product_id)
+
+    @property
+    def button_pressed(self) -> Optional[str]:
+        """Last pressed button."""
+        return self.data["button_pressed"]
+
+    @classmethod
+    def _get_filter_type(cls, product_id: str) -> FilterType:
+        ft = cls._filter_type_cache.get(product_id, None)
+        if ft is None:
+            for filter_re, filter_type in FILTER_TYPE_RE:
+                if filter_re.match(product_id):
+                    ft = cls._filter_type_cache[product_id] = filter_type
+                    break
+        return ft
+
+    def __repr__(self) -> str:
+        s = (
+            "<AirPurifierMiotStatus power=%s, "
+            "aqi=%s, "
+            "average_aqi=%s, "
+            "temperature=%s, "
+            "humidity=%s%%, "
+            "fan_level=%s, "
+            "mode=%s, "
+            "led=%s, "
+            "led_brightness=%s, "
+            "buzzer=%s, "
+            "child_lock=%s, "
+            "favorite_level=%s, "
+            "filter_life_remaining=%s, "
+            "filter_hours_used=%s, "
+            "use_time=%s, "
+            "purify_volume=%s, "
+            "motor_speed=%s, "
+            "filter_rfid_product_id=%s, "
+            "filter_rfid_tag=%s, "
+            "filter_type=%s>"
+            % (
+                self.power,
+                self.aqi,
+                self.average_aqi,
+                self.temperature,
+                self.humidity,
+                self.fan_level,
+                self.mode,
+                self.led,
+                self.led_brightness,
+                self.buzzer,
+                self.child_lock,
+                self.favorite_level,
+                self.filter_life_remaining,
+                self.filter_hours_used,
+                self.use_time,
+                self.purify_volume,
+                self.motor_speed,
+                self.filter_rfid_product_id,
+                self.filter_rfid_tag,
+                self.filter_type,
+            )
+        )
+        return s
+
+    def __json__(self):
+        return self.data
+
+
+class AirPurifierMiot(MiotDevice):
```

Comment:
Just curious if you've tried out Property piid: 9 (motor1-set-speed): (int32, unit: None) (acc: ['read', 'write'], value-list: [], value-range: [0, 3000, 1]).

Can it be used to set motor rpm speed directly instead of using favorite_level steps? 

Would be useful for better speed control with hysteresis instead of jumps in motor speed. Probably better for the motor, too.

And anyone using the library can map it better to their apps.